arches/app/models/resource.py

```diff
@@ -901,9 +902,15 @@ def get_relations(
                 and str(resourceinstancefrom_graphid) in readable_graphids
             ):
                 try:
-                    preflabel = get_preflabel_from_valueid(
-                        relation["relationshiptype"], lang
-                    )
+                    if f'{relation["relationshiptype"]}{lang}' not in preflabel_lookup:
+                        preflabel_lookup[f'{relation["relationshiptype"]}{lang}'] = (
+                            get_preflabel_from_valueid(
+                                relation["relationshiptype"], lang
+                            )
+                        )
+                    preflabel = preflabel_lookup[
+                        f'{relation["relationshiptype"]}{lang}'
+                    ]
```

Comment:
If this were reordered a bit, there wouldn't be a need to check the dictionary for the value twice (once to check if the value is not in the preflable_lookup and a second time to retrieve it): 
```
try: 
    preflabel = preflabel_lookup[f'{relation["relationshiptype"]}{lang}']
except KeyError:
    preflabel = get_preflabel_from_valueid(relation["relationshiptype"], lang)
    preflabel_lookup[f'{relation["relationshiptype"]}{lang}'] = preflabel
```

Reply:
Given this code below:
```.py
    relation["relationshiptype_label"] = preflabel["value"] or ""
except:
    relation["relationshiptype_label"] = (
        relation["relationshiptype"] or ""
    )
```
while there's some ambiguity as to which exceptions it was meant to handle, at a minimum there's an anticipation that either `preflabel["value"]` could raise an error (`KeyError`?) or that `relation["relationshiptype"]` is `None` (causing perhaps an ES 404 error downstream in `get_preflabel_from_valueid`).

Reply:
Here's the error reproduced (that the original `except` was meant to handle, as one outcome)
```
Internal Server Error: /resource/related/b6d2d1e1-9db8-4787-92ca-048b34bf6d66
Traceback (most recent call last):
  File "/Users/galenmancino/repos/arches/arches/app/models/resource.py", line 905, in get_related_resources
    preflabel = preflabel_lookup[f'{relation["relationshiptype"]}{lang}']
                ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'Noneen'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/galenmancino/repos/venv_312_arches7.6/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/galenmancino/repos/venv_312_arches7.6/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/galenmancino/repos/venv_312_arches7.6/lib/python3.12/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/galenmancino/repos/venv_312_arches7.6/lib/python3.12/site-packages/django/utils/decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/galenmancino/repos/arches/arches/app/utils/decorators.py", line 109, in wrapper
    return function(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/galenmancino/repos/venv_312_arches7.6/lib/python3.12/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/galenmancino/repos/arches/arches/app/views/resource.py", line 1064, in get
    ret = resource.get_related_resources(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/galenmancino/repos/arches/arches/app/models/resource.py", line 908, in get_related_resources
    preflabel = get_preflabel_from_valueid(relation["relationshiptype"], lang)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/galenmancino/repos/arches/arches/app/models/concept.py", line 1734, in get_preflabel_from_valueid
    if concept_label["found"]:
       ~~~~~~~~~~~~~^^^^^^^^^
KeyError: 'found'
```